applications/wikipedia/upload.py

```diff
@@ -0,0 +1,63 @@
+from pathlib import Path
+import time
+from modal import Image, Stub, Volume, Secret
+
+MODEL_ID = "BAAI/bge-small-en-v1.5"
+MODEL_SLUG = MODEL_ID.split("/")[-1]
+
+BATCH_SIZE = 512
+DOCKER_IMAGE = (
+    "ghcr.io/huggingface/text-embeddings-inference:86-0.4.0"  # Ampere 86 for A10s.
+    # "ghcr.io/huggingface/text-embeddings-inference:0.4.0" # Ampere 80 for A100s.
+    # "ghcr.io/huggingface/text-embeddings-inference:0.3.0"  # Turing for T4s.
+)
+dataset_name = "wikipedia"
+volume = Volume.persisted("embedding-wikipedia")
+cache_dir = "/data"
+data_dir = f"{cache_dir}/{dataset_name}"
+DATA_PATH = Path(data_dir)
+
+dataset_name = "567-labs/upload-test-benchmark"
+dataset_file = "wiki-embeddings.parquet"
+
+
+stub = Stub("embeddings")
+
+
+@stub.function(
+    image=Image.debian_slim().pip_install(
+        "datasets", "pyarrow", "tqdm", "hf_transfer", "huggingface_hub"
+    ),
+    volumes={cache_dir: volume},
+    _allow_background_volume_commits=True,
+    timeout=84600,
+    secret=Secret.from_name("huggingface-credentials"),
+)
+def upload_dataset():
+    from huggingface_hub import HfApi
+    import os
+    api = HfApi(token=os.environ["HUGGINGFACE_TOKEN"])
+    api.create_repo(repo_id=dataset_name, private=False, repo_type="dataset",exist_ok=True)
+
+
+    print(f"Pushing to hub {dataset_name}")
+    # dataset = load_dataset(f"{cache_dir}/wikipedia")
+    start = time.perf_counter()
+    # dataset.push_to_hub(dataset_name)
+    api.upload_folder(
+        folder_path=f"{cache_dir}/wikipedia",
```

Comment:
Can you update this to the checkpoints and then try a 25% batch and see how long it takes.

Reply:
```
  step b1a6b6e1eff0e9e0fa2a6d579b8f38b93af9bec1e499c724818f11cc2186b2c0 completed (still 1 to go).
data-00124-of-00127.arrow: 512MB [00:01, 512MB/s]                           
data-00125-of-00127.arrow: 512MB [00:10, 51.1MB/s]                          
data-00126-of-00127.arrow: 512MB [00:01, 401MB/s]                           
100%|██████████| 3/3 [00:12<00:00,  4.27s/it]
  step c9c3d34719f6b51dac2a4c29e9b3b20a440bb1f40a7fa669e18185dbb1aacf9e completed (still 0 to go).
All commits have been pushed.
PR is now open for reviews.
PR has been automatically merged (`merge_pr=True` was passed).
Uploaded in 758.460917796s
```

It took about 12.5 minutes to upload the entire dataset. I've found that upload time doesn't scale as much as the absolute size of the dataset though, 10% of dataset took 10 minutes while 25% took 12 minutes. We might be able to stay < 30 mins for uploading the entire dataset of embeddings ( optimistically )